### DISCUSSION TITLE
Incremental Static Regeneration seems to regenerate pages that were not visited.

### DISCUSSION THREAD

**Discussion by labago** [vercel/next.js#38577] [Help]
We have a website that runs off of Contentful content, over 600 pages. I have a console log that runs anytime our [[...slug]].tsx page is built (so whenever one of those 600 pages is built). You can see each console log for each page being built when we run a local build. After the build, and running "next start" locally, I was testing the static regeneration with revalidate set to 60. I would expect that, after I load the root page "/", that after 60 seconds pass and I load the page again, I would see that console log run that shows the page is being regenerated. This is true, I do see that console log, but I ALSO see logs for about 10-15 other seemingly random pages being rebuilt, even though I did not visit those pages. Why is this occurring? Shouldn't NextJS only be regenerating the page that I just visited, and not other random pages? Services like Contentful have rate limits, so if pages are just none stop being regenerated, we hit those limits seemingly all the time, which causes other issues inside the application. I am looking to see if anyone else has run into this and if it is expected NextJS behavior. According to their documentation on ISR, it does not seem to be.

Thanks!

**Comment by pzi** [comment]
You might want to look at the `prefetch` property of `next/link`: https://nextjs.org/docs/api-reference/next/link - it will prefetch everything in the viewport and on hover.
